src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -352,6 +352,7 @@ public void setTokens(Collection<Token> tokens)
         setGossipTokens(localTokens);
         tokenMetadata.updateNormalTokens(tokens, FBUtilities.getBroadcastAddressAndPort());
         setMode(Mode.NORMAL, false);
+        invalidateLocalRanges();
```

Comment:
Will this cause a `DiskBoundaryManager` invalidation in CFS where it didn't before? Was that a bug?

Reply:
The disk boundary manager would recognize the change in ring version and apply the change the next time the disk boundaries are needed, i.e. on the next flush or compaction. The invalidation here is in fact no-op for it.

Memtables, on the other hand, need a signal to trigger a flush to maintain per-shard balance.